src/main/java/io/shelang/aghab/service/link/LinksServiceImpl.java

```diff
@@ -71,17 +85,23 @@ private Links initCreation(LinkCreateDTO dto) {
   @Override
   @Transactional
   public LinksDTO create(LinkCreateDTO dto) {
+    User user = usersRepository.findByIdOptional(userId).orElseThrow(NotFoundException::new);
     byte retry = 0;
     if (dto.getHash() != null) retry = MAX_RETRY_COUNT - 1;
     Links link = initCreation(dto);
     persistAndRetry(link, retry);
+    persistLinkUser(link, user);
```

Comment:
Would it be possible to make persistLinkUser call after persistAndRetry and handling exceptions?

Reply:
Could you explain more? The persistAndRetry also could it throws an error if it could not persist the link by the retrying policy

Reply:
Ok, I get it